### PR TITLE
Feature add seo priority setting

### PR DESCRIPTION
**Pull request by nelsonchuah**
Hi Dom,

Could you help review the additional feature added to sitemap extension, enabling user to set page's priority.

We'll add additional `<priority>0.8</priority>` property to page in sitemap.xml based on the priority set by user in page

reference doc
- https://www.sitemaps.org/protocol.html#prioritydef

### CODE REVIEW

File: services/GoogleSitemapService.cfc [hunk 1]
```diff
@@ -94,13 +105,15 @@ component {
 		sitemap.append( newline & '<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">' );
 
 		for ( var page in arguments.pages ) {
-			loc     = siteRootUrl.reReplace( "/$", "" ) & page._hierarchy_slug.reReplace( "(.)/$", "\1.html" );
-			lastmod = DateFormat( page.datemodified, "yyyy-mm-dd" );
+			loc      = siteRootUrl.reReplace( "/$", "" ) & page._hierarchy_slug.reReplace( "(.)/$", "\1.html" );
+			lastmod  = DateFormat( page.datemodified, "yyyy-mm-dd" );
+			priority = _getPriorityRange( page.sitemap_priority );
```
Comment (DominicWatson) [inline services/GoogleSitemapService.cfc:111]: I think this variable requires a var scope @nelsonchuah ?

File: services/GoogleSitemapService.cfc [hunk 2]
```diff
@@ -145,22 +158,46 @@ component {
 		};
 	}
 
-	private function _addChildPages( required array haveAccessPages, required array childPages, string parentSearchEngineAccess, string parentAccessRestriction ) {
+	private struct function _getSitemapPriorityForPage( required string pageId ) {
```
Comment (DominicWatson) [inline services/GoogleSitemapService.cfc]: Any good reason this doesn't just return the string of the priority? Can see it used only in one place and it immediately gets the "sitemap_priority" key from the returned struct.

**Review by DominicWatson — CHANGES_REQUESTED** [review]
Hey @nelsonchuah - just a couple of minor things here - one possible issue with var scoping.

**Comment by nelsonchuah** [comment]
hi Dom,

Have updated the missing var scoping for priority variable and updated return type for _getSitemapPriorityForPage() function

Thank you
Nelson

File: services/GoogleSitemapService.cfc [hunk 3]
```diff
@@ -65,12 +67,21 @@ component {
 				}
 			}
 
+			if ( page.sitemap_priority=="inherit" ) {
+				if ( !structKeyExists( inheritedPageSitemapPriority, page.parent_page ) ) {
+					page.sitemap_priority = _getSitemapPriorityForPage( page.id ).sitemap_priority;
```
Comment (DominicWatson) [inline services/GoogleSitemapService.cfc]: This needs amending to fit with the change to the method signature

**Review by DominicWatson — CHANGES_REQUESTED** [review]
Nearly there...

File: services/GoogleSitemapService.cfc (hunk 3 above)
Comment (nelsonchuah) [reply services/GoogleSitemapService.cfc]: Hi Dom, 

Have updated the method call

Thank you
Nelson

**Comment by DominicWatson** [comment]
Nice one, merging and releasing as 1.1.0

**Pull request merged**